packages/remix-dev/config.ts

```diff
@@ -342,20 +375,41 @@ export async function readConfig(
     }
   }
 
-  let customServerEntryPoint = appConfig.server;
-  let serverBuildTarget: ServerBuildTarget | undefined =
-    appConfig.serverBuildTarget;
-  let serverModuleFormat: ServerModuleFormat =
-    appConfig.serverModuleFormat || "cjs";
-  let serverPlatform: ServerPlatform = appConfig.serverPlatform || "node";
-  switch (appConfig.serverBuildTarget) {
-    case "cloudflare-pages":
-    case "cloudflare-workers":
-    case "deno":
-      serverModuleFormat = "esm";
-      serverPlatform = "neutral";
-      break;
+  let isCloudflareRuntime = ["cloudflare-pages", "cloudflare-workers"].includes(
+    appConfig.serverBuildTarget ?? ""
+  );
+  let isDenoRuntime = appConfig.serverBuildTarget === "deno";
+
+  let serverBuildPath = resolveServerBuildPath(rootDirectory, appConfig);
+  let serverBuildTarget = appConfig.serverBuildTarget;
+  let serverBuildTargetEntryModule = `export * from ${JSON.stringify(
+    serverBuildVirtualModule.id
+  )};`;
+  let serverConditions = appConfig.serverConditions;
+  let serverDependenciesToBundle = appConfig.serverDependenciesToBundle || [];
+  let serverEntryPoint = appConfig.server;
+  let serverMainFields = appConfig.serverMainFields;
+  let serverMinify = appConfig.serverMinify;
+  let serverModuleFormat = appConfig.serverModuleFormat || "cjs";
+  let serverPlatform = appConfig.serverPlatform || "node";
+  if (isCloudflareRuntime) {
+    serverConditions ??= ["worker"];
+    serverDependenciesToBundle = "all";
+    serverMainFields ??= ["browser", "module", "main"];
+    serverMinify ??= true;
+    serverModuleFormat = "esm";
+    serverPlatform = "neutral";
+  }
+  if (isDenoRuntime) {
+    serverConditions ??= ["deno", "worker"];
+    serverDependenciesToBundle = "all";
+    serverMainFields ??= ["module", "main"];
+    serverModuleFormat = "esm";
+    serverPlatform = "neutral";
   }
+  serverMainFields ??=
```

Comment:
This should be 
```js
serverMainFields = appConfig.serverMainFields ?? serverMainFields ?? serverModuleFormat === "esm" ? ["module", "main"] : ["main", "module"];
```

Reply:
Hmm that's weird 🤔

Because in the declaration of the variable, I'm setting it to `appConfig.serverMainFields` & I'm only overriding it in the `if`-statements if it doesn't have a value yet
Same for the `esm` check: only doing that if it doesn't have a value yet